### PR TITLE
Parse `docker prune` Output

### PR DESCRIPTION
**Pull request by alexyaang**
TODO: 

- [x] More testing

### CODE REVIEW

File: packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+export type PruneResult = {
+    spaceReclaimed: number;
+    resources: string[];
+};
+
+export function parsePruneLikeOutput(output: string, options: PruneParseOptions): PruneResult {
+    const deletedResources: string[] = [];
+    let spaceReclaimed: number = 0;
+
+    const lines = asIds(output);
```
Comment (bwateratmsft) [inline packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts:23]: `asIds` isn't meant to be used as a line-breaking feature. In fact, if you use `'foo'.match(/regex/g)` you might not need to break things up by line at all, just run it through the two regex (one for matching ID's/names of resources, one for matching the reclaimed space.

Something like this:

```typescript
const resourceMatches = output.match(/^\w+$/igm); // the G flag means global, gimme all matches. The M flag means multiline, which says to treat ^ and $ as the beginning and end of lines, not the whole string

const spaceReclaimedMatches = /Total space reclaimed: ([\w\s]+)/i.exec(output);

// Continue with getting the group out, tryParseSize, etc.
```

Comment (alexyaang) [reply packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts:23]: yeah that makes sense

File: packages/vscode-container-client/src/clients/DockerClientBase/DockerClientBase.ts [hunk 2]
```diff
@@ -518,8 +519,17 @@ export abstract class DockerClientBase extends ConfigurableClient implements ICo
         output: string,
         strict: boolean,
     ): Promise<PruneImagesItem> {
-        // TODO: Parse output for prune info
-        return Promise.resolve({});
+        const deletedImageLineRegex = /^deleted: sha256:\s*(\w+)$/igm;
+
+        const deletedImages = parsePruneLikeOutput(output, {
+            spaceReclaimedRegex: undefined,
```
Comment (bwateratmsft) [inline packages/vscode-container-client/src/clients/DockerClientBase/DockerClientBase.ts:525]: All three calls to `parsePruneLikeOutput` have undefined for `spaceReclaimedRegex`, seems like you can just remove the option entirely.

File: packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    let deletedResources: string[] = [];
+    let spaceReclaimed: number = 0;
+
+    // match resources
+    deletedResources = output.match(resourceRegex) || [];
+
+    // remove the space reclaimed line if it exists
+    if (deletedResources.length > 0) {
+        deletedResources.pop();
+    }
```
Comment (bwateratmsft) [inline packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts:37]: You should craft the regexp so this step isn't needed. i.e. `LineRegex` or `resourceRegex` should not match `"Total space reclaimed: 1234 MB"`

Comment (alexyaang) [reply packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts:37]: you're also right here as well. through some more testing, I realized this step is actually messes things up, going to remove iti.

**Review by bwateratmsft — APPROVED** [review]

**Pull request merged**